ext/mysqlnd/mysqlnd_ps_codec.c

```diff
@@ -237,7 +240,9 @@ ps_fetch_time(zval * zv, const MYSQLND_FIELD * const field, const unsigned int p
 		t.time_type = MYSQLND_TIMESTAMP_TIME;
 	}
 
-	if (field->decimals > 0 && field->decimals < 7) {
+	if (field->decimals > 0 && (field->decimals <= MYSQLND_SEC_PART_DIGITS ||
+                             (field->decimals == MYSQLND_AUTO_SEC_PART_DIGITS && t.second_part))) {
+    uint8_t decimals= (field->decimals == MYSQLND_AUTO_SEC_PART_DIGITS) ? MYSQLND_SEC_PART_DIGITS : field->decimals;
```

Comment:
I think you should probably replace `field->decimals` with `decimals` in `zend_strpprintf` below.